agent/hcp/telemetry/otel_exporter.go

```diff
@@ -0,0 +1,66 @@
+package telemetry
+
+import (
+	"context"
+
+	hcpclient "github.com/hashicorp/consul/agent/hcp/client"
+
+	"github.com/hashicorp/go-multierror"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// OTELExporter is a custom implementation of a OTEL Metrics SDK metrics.Exporter.
+// The exporter is used by a OTEL Metrics SDK PeriodicReader to export aggregated metrics.
+// This allows us to use a custom client - HCP authenticated MetricsClient.
+type OTELExporter struct {
+	client   hcpclient.MetricsClient
+	endpoint string
+}
+
+// NewOTELExporter returns a configured OTELExporter
+func NewOTELExporter(client hcpclient.MetricsClient, endpoint string) *OTELExporter {
+	return &OTELExporter{
+		client:   client,
+		endpoint: endpoint,
+	}
+}
+
+// Temporality returns the Cumulative temporality for metrics aggregation.
+// Telemetry Gateway stores metrics in Prometheus format, so use Cummulative aggregation as default.
+func (e *OTELExporter) Temporality(_ metric.InstrumentKind) metricdata.Temporality {
+	return metricdata.CumulativeTemporality
+}
+
+// Aggregation returns the Aggregation to use for an instrument kind.
+func (e *OTELExporter) Aggregation(kind metric.InstrumentKind) aggregation.Aggregation {
+	switch kind {
+	case metric.InstrumentKindObservableGauge:
+		return aggregation.LastValue{}
+	case metric.InstrumentKindHistogram:
+		return aggregation.ExplicitBucketHistogram{
+			Boundaries: []float64{0, 5, 10, 25, 50, 75, 100, 250, 500, 750, 1000, 2500, 5000, 7500, 10000},
+			NoMinMax:   false,
+		}
+	}
+	// for metric.InstrumentKindCounter and others, default to sum.
+	return aggregation.Sum{}
+}
+
+// Export serializes and transmits metric data to a receiver.
+func (e *OTELExporter) Export(ctx context.Context, metrics metricdata.ResourceMetrics) error {
+	if len(metrics.ScopeMetrics) == 0 || len(metrics.ScopeMetrics[0].Metrics) == 0 {
+		return nil
+	}
+
+	otlpMetrics, merr := transformOTLP(&metrics)
```

Comment:
You _might_ be able to just use the proto unmarshaller but let's rule it out. I'd rather us not have to use our own crafted converters if possible.

Reply:
We might want to check this merr before we export Metrics? Preferably we should return known error types vs mutlierrors so we can switch and act on the type.

Reply:
I've updated this to remove that error, since we plan on sending a metric for errors

Reply:
Re - proto unmarshaleller

I'd like to keep the current implementation, rationale:

1. I saw that the Gateway uses a [ProtoUnmarshaler](https://pkg.go.dev/go.opentelemetry.io/collector/pdata@v0.66.0/pmetric#ProtoUnmarshaler) provided by the pdata (internal representation of metrics). However, I don't see support for marshal/unmarshal in the metrics SDK [here](https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric@v0.38.1/metricdata#) and protobufs [here](https://pkg.go.dev/go.opentelemetry.io/proto/otlp@v0.19.0/metrics/v1). 
2. As we discussed, it might be a good idea to have this to prevent a single metric transformation error from sending all metrics